lib/ts/framework/custom/framework.ts

```diff
@@ -0,0 +1,203 @@
+/* Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import type { HTTPMethod } from "../../types";
+import { makeDefaultUserContextFromAPI, normaliseHttpMethod } from "../../utils";
+import { BaseRequest } from "../request";
+import { BaseResponse } from "../response";
+import SuperTokens from "../../supertokens";
+
+type RequestInfo = {
+    url: string;
+    method: HTTPMethod;
+    headers: Headers;
+    cookies: Record<string, string>;
+    query: Record<string, string>;
+    getJSONBody: () => Promise<any>;
+    getFormBody: () => Promise<any>;
+};
+
```

Comment:
custom frameworks also need to provide an impl for verifySession middleware, which is in the session recipe.